File: python/test_color.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals
from color import *

english = "english"
chinese = "中文"


def fix_width(s, length=10):
    return s.ljust(length)


name_list = ["black", "red", "green", "yellow", "blue", "magenta", "cyan", "white"]
fg_list = [black, red, green, yellow, blue, magenta, cyan, white]
bg_list = [black_bg, red_bg, green_bg, yellow_bg, blue_bg, magenta_bg, cyan_bg, white_bg]

for name, fg, bg in zip(name_list, fg_list, bg_list):
    print("%s \t %s \t %s \t %s \t %s" % (
        fix_width(name),
        fg(english), fg(chinese),
        bg(english), bg(chinese),
    ))

print("")

name_list = ["br_black", "br_red", "br_green", "br_yellow", "br_blue", "br_magenta", "br_cyan", "br_white"]
fg_list = [br_black, br_red, br_green, br_yellow, br_blue, br_magenta, br_cyan, br_white]
bg_list = [br_black_bg, br_red_bg, br_green_bg, br_yellow_bg, br_blue_bg, br_magenta_bg, br_cyan_bg, br_white_bg]

for name, fg, bg in zip(name_list, fg_list, bg_list):
    print("%s \t %s \t %s \t %s \t %s" % (
        fix_width(name),
        fg(english), fg(chinese),
        bg(english), bg(chinese),
    ))

print("")

name_list = ["black_hl", "red_hl", "green_hl", "yellow_hl", "blue_hl", "magenta_hl", "cyan_hl", "white_hl"]
fg_list = [black_hl, red_hl, green_hl, yellow_hl, blue_hl, magenta_hl, cyan_hl, white_hl]

for name, fg in zip(name_list, fg_list):
    print("%s \t %s \t %s" % (
        fix_width(name),
        fg(english), fg(chinese),
    ))

print("")

name_list = ["bold", "italic", "underline", "blink", "strike"]
style_list = [bold, italic, underline, blink, strike]

print("%s \t %s \t %s" % (fix_width("normal"), english, chinese))
for name, style in zip(name_list, style_list):
    print("%s \t %s \t %s" % (
        fix_width(name),
        style("english"),
        style("中文"),
    ))

print("")
print(red(br_yellow_bg("use functions composed together")))

File: nodejs/test_color.js
const color = require('./color')

const english = 'english'
const chinese = '中文'

function fix_width(s){
    const width = 10
    return s.length >= width ? s : s + new Array(width - s.length).join(" ");
}

name_list = ["black", "red", "green", "yellow", "blue", "magenta", "cyan", "white"]

fg_list = [color.black, color.red, color.green, color.yellow, color.blue, color.magenta, color.cyan, color.white]

bg_list = [color.black_bg, color.red_bg, color.green_bg, color.yellow_bg, color.blue_bg, color.magenta_bg, color.cyan_bg, color.white_bg]

hl_list = [color.black_hl, color.red_hl, color.green_hl, color.yellow_hl, color.blue_hl, color.magenta_hl, color.cyan_hl, color.white_hl]

name_list.forEach(function(v, i){
    console.log(fix_width(v) + fg_list[i](english) + '\t' 
        + fg_list[i](chinese) + '\t' 
        + bg_list[i](english) + '\t' 
        + bg_list[i](chinese) + '\t'
        + hl_list[i](english) + '\t'
        + hl_list[i](chinese)
    )
})

console.log()

name_list = ['bold', 'italic', 'underline', 'blink', 'strike']
style_list = [color.bold, color.italic, color.underline, color.blink, color.strike]

console.log(fix_width('normal') + english + '\t' + chinese)
name_list.forEach(function(v, i){
    console.log(fix_width(v) + style_list[i](english) + '\t'
        + style_list[i](chinese)
    )
})
